pybamm/models/submodels/electrolyte_conductivity/integrated_conductivity.py

```diff
@@ -0,0 +1,173 @@
+        return pybamm.log(x)
+
+    def get_coupled_variables(self, variables):
+        c_e_av = variables["X-averaged electrolyte concentration"]
+
+        i_boundary_cc_0 = variables["Leading-order current collector current density"]
+        c_e_n = variables["Negative electrolyte concentration"]
+        c_e_s = variables["Separator electrolyte concentration"]
+        c_e_p = variables["Positive electrolyte concentration"]
+        c_e_n0 = pybamm.boundary_value(c_e_n, "left")
+
+        delta_phi_n_av = variables[
+            "X-averaged negative electrode surface potential difference"
+        ]
+        phi_s_n_av = variables["X-averaged negative electrode potential"]
+
+        tor_n = variables["Negative electrolyte tortuosity"]
+        tor_s = variables["Separator tortuosity"]
+        tor_p = variables["Positive electrolyte tortuosity"]
+
+        T_av = variables["X-averaged cell temperature"]
+        T_av_n = pybamm.PrimaryBroadcast(T_av, "negative electrode")
+        T_av_s = pybamm.PrimaryBroadcast(T_av, "separator")
+        T_av_p = pybamm.PrimaryBroadcast(T_av, "positive electrode")
+
+        param = self.param
+        l_n = param.l_n
+        l_p = param.l_p
+        x_n = pybamm.standard_spatial_vars.x_n
+        x_s = pybamm.standard_spatial_vars.x_s
+        x_p = pybamm.standard_spatial_vars.x_p
+        x_n_edge = pybamm.standard_spatial_vars.x_n_edge
+        x_p_edge = pybamm.standard_spatial_vars.x_p_edge
+
+        chi_av = param.chi(c_e_av)
+        chi_av_n = pybamm.PrimaryBroadcast(chi_av, "negative electrode")
+        chi_av_s = pybamm.PrimaryBroadcast(chi_av, "separator")
+        chi_av_p = pybamm.PrimaryBroadcast(chi_av, "positive electrode")
+
+        # electrolyte current
+        i_e_n = i_boundary_cc_0 * x_n / l_n
+        i_e_s = pybamm.PrimaryBroadcast(i_boundary_cc_0, "separator")
+        i_e_p = i_boundary_cc_0 * (1 - x_p) / l_p
+        i_e = pybamm.Concatenation(i_e_n, i_e_s, i_e_p)
+
+        i_e_n_edge = i_boundary_cc_0 * x_n_edge / l_n
+        i_e_s_edge = pybamm.PrimaryBroadcastToEdges(i_boundary_cc_0, "separator")
+        i_e_p_edge = i_boundary_cc_0 * (1 - x_p_edge) / l_p
+
+        # electrolyte potential
+        indef_integral_n = (
+            pybamm.IndefiniteIntegral(
+                i_e_n_edge / (param.kappa_e(c_e_n, T_av_n) * tor_n), x_n
+            )
+            * param.C_e
+            / param.gamma_e
+        )
+        indef_integral_s = (
+            pybamm.IndefiniteIntegral(
+                i_e_s_edge / (param.kappa_e(c_e_s, T_av_s) * tor_s), x_s
+            )
+            * param.C_e
+            / param.gamma_e
+        )
+        indef_integral_p = (
+            pybamm.IndefiniteIntegral(
+                i_e_p_edge / (param.kappa_e(c_e_p, T_av_p) * tor_p), x_p
+            )
+            * param.C_e
+            / param.gamma_e
+        )
+
+        integral_n = indef_integral_n - pybamm.boundary_value(indef_integral_n, "left")
```

Comment:
left boundary value of an indefinite integral should always be zero by definition (and implementation) but worth checking

Reply:
I checked, and maybe it evaluates to zero (actually it should), but is not zero by definition. The definition is
```
<bound method UnaryOperator.evaluate of BoundaryValue(-0x33bf8d7889aadb3a, boundary value, children=["* integrated w.r.t x_n on ['negative electrode'](3.0 * x_n ** 2.0)"], domain=['current collector'], auxiliary_domains={})>
```